src/client/profile-provider.ts

```diff
@@ -368,26 +368,34 @@ export class ProfileProvider {
     }
     const result: Record<string, string> = {};
 
+    //Resolve parameters defined in super.json
     for (const [key, value] of Object.entries(superJsonParameters)) {
-      // If there is value in env use it
-      let envValue: string | undefined = undefined;
-      if (value.startsWith('$')) {
-        envValue = process.env[value.substr(1)];
-      }
-      if (envValue !== undefined) {
-        result[key] = envValue;
-      } else {
-        // If we have defualt in provider.json use it
-        const providerJsonDefault = providerJsonParameters.find(
-          parameter => parameter.name === key && parameter.default
-        )?.default;
-        if (providerJsonDefault) {
-          result[key] = providerJsonDefault;
-          // Use value as is
-        } else {
-          result[key] = value;
+      const providerJsonParameter = providerJsonParameters.find(
+        parameter => parameter.name === key
+      );
+      if (providerJsonParameter) {
+        //If value name and prepared value equals we are dealing with unset env
+        if (
+          prepareProviderParameters(providerJson.name, [providerJsonParameter])[
```

Comment:
can you do this before the loop and just access it here so you don't call it (in worst case) every loop and also it will look better